File: s05_OOP/OOPProject/app/src/main/java/com/omeryildizce/oopproject/Mathematics.java
package com.omeryildizce.oopproject;

import java.util.Arrays;

public class Mathematics {
    // Static polymorphism
    public int sum(){
        return 0;
    }

    public int sum(int x, int y){
        return x + y;
    }

    public int sum(int... args){
        int sum = 0;
        for(int x: args){
            sum+=x;
        }
        System.out.println(sum);
        return sum;
    }


}

File: s18_OOP/OOP/app/src/main/java/com/omeryildizce/oop/Mathematics.kt
package com.omeryildizce.oop

class Mathematics {
    fun sum(): Int{
        return 0
    }
    fun sum(x:Int, y:Int):Int{
        return x + y
    }
    fun sum(x:Int, y:Int, z:Int):Int{
        return x + y + z
    }
    fun sum(vararg x:Int) :Int{
        var sum = 0
        for (i in x){
            sum+=i
        }
        return sum
    }
}